lib/Target/CBackend/CBackend.cpp

```diff
@@ -870,13 +871,13 @@ bool CWriter::printConstantString(Constant *C, enum OperandContext Context) {
 
 // TODO copied from CppBackend, new code should use raw_ostream
 static inline std::string ftostr(const APFloat &V) {
-  std::string Buf;
+  std::stringstream ss;
```

Comment:
By default, a new created `std::stringstream` uses whatever global process C++ locale is set with [`std::locale::global`](https://en.cppreference.com/w/cpp/locale/locale/global). To avoid weird locale corner cases, it probably make sense to explicitly specify the used C locale with:
```suggestion
  std::stringstream ss;
  ss.imbue(std::locale::classic());
```

Reply:
Good idea, fixing it.